src/subscription_manager/certdirectory.py

```diff
@@ -330,6 +330,21 @@ def list_for_product(self, product_id):
                     entitlements.append(cert)
         return entitlements
 
+    def list_for_product_name(self, product_name, case_sensitive=True):
+        """
+        Returns all entitlement certificates providing access to the given
+        product by name.
+        """
+        entitlements = []
+        for cert in self.list():
+            for cert_product in cert.products:
+                if product_name == cert_product.name:
```

Comment:
I believe that the logic following should be following to avoid useless test, when `case_sensitive` is `False`:

```python
if case_sensitive is True:
    if product_name == cert_product.name:
        entitlements.append(cert)
else:
    if product_name.casefold() == cert_product.name.casefold():
        entitlements.append(cert)
```

Reply:
I've made the requested change.